fe/fe-core/src/main/java/com/starrocks/mysql/privilege/PrivTable.java

```diff
@@ -60,8 +73,9 @@ public PrivEntry addEntry(PrivEntry newEntry, boolean errOnExist, boolean errOnN
             if (errOnNonExist) {
                 throw new DdlException("User " + newEntry.getUserIdent() + " does not exist");
             }
+            UserIdentity newUser = newEntry.getUserIdent();
+            List<PrivEntry> entries = map.computeIfAbsent(newUser, k -> new ArrayList<>());
             entries.add(newEntry);
-            Collections.sort(entries);
```

Comment:
Why delete the sort code?

Reply:
If you delete the sort code here. The items returned by getReadOnlyIteratorByUser() should be ordered. Because some check code is based on the order, such as login.

Reply:
If you delete the sort code here. The items returned by getReadOnlyIteratorByUser() should be ordered. Because some check code is based on the order, such as login.

Reply:
All users remain sorted using the Comparator from line 50.
Since we already extract the UserIdentity as the key of the map, the value of the map which represent all entries of a certain user will no longer required sorted.

Reply:
https://github.com/StarRocks/starrocks/blob/main/fe/fe-core/src/main/java/com/starrocks/mysql/privilege/GlobalPrivEntry.java#L112
The user is sorted first by host, better to be same as before. And this is the mysql's Restrictions.